### PR TITLE
[zh-cn]: update the translation of SharedArrayBuffer overview

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

Is it possible to consider translating `worker thread(s)` to `工作线程`？I noticed that some Chinese technical articles or blogs translated it as so,  and the `System Worker Thread`, a similar concept which some different, in some [Microsoft documentation](https://learn.microsoft.com/zh-cn/windows-hardware/drivers/kernel/system-worker-threads).

### Related issues and pull requests

#17175

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 1]
```diff
@@ -1,98 +1,129 @@
...
-```
+### API 可用性
 
-```js example-good
-var sab = new SharedArrayBuffer(1024);
-```
+根据是否采取了上述安全措施，各类内存共享 API 具有不同的可用性：
+
+- `Atomics` 对象总是可用的。
+- `SharedArrayBuffer` 对象在原则上始终可用，但遗憾的是，除非设置了前面提到的两个头，否则其在全局对象上的构造函数是隐藏的，以兼容 web 内容。有望在未来移除这个限制。尽管如此，仍然可以用 [`WebAssembly.Memory`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory) 来获取实例。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:72]:
```suggestion
- `SharedArrayBuffer` 对象在原则上始终可用，但遗憾的是，除非设置了前面提到的两个头，否则其在全局对象上的构造函数是隐藏的，这是为了兼容 web 内容。这个限制有望在未来被移除。尽管如此，仍然可以用 [`WebAssembly.Memory`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory) 来获取实例。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 2]
```diff
@@ -1,98 +1,129 @@
...
+
+WebAssembly Threads 提案还定义了一套新的[原子](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#atomic-memory-accesses)指令。就像 `SharedArrayBuffer` 及其方法始终可用（并且只有在设置了新头部的情况下，才允许线程间共享）一样，WebAssembly 原子指令也是始终可用的。
+
+### 增长 SharedArrayBuffer
+
+`SharedArrayBuffer` 对象可以通过在调用 {{jsxref("SharedArrayBuffer/SharedArrayBuffer", "SharedArrayBuffer()")}} 时包含 `maxByteLength` 选项来使其可增长。你可以通过访问 `SharedArrayBuffer` 的 {{jsxref("SharedArrayBuffer/growable", "growable")}} 和 {{jsxref("SharedArrayBuffer/maxByteLength", "maxByteLength")}} 属性来分别查询其是否可增长以及其最大大小。你还可以通过调用 {{jsxref("SharedArrayBuffer/grow", "grow()")}} 为一个可增长的 `SharedArrayBuffer` 分配新的大小。新字节被初始化为 0。
+
+这些特性令增长 `SharedArrayBuffer` 更为高效——否则，你必须将创建一个新大小的缓冲区副本。它还使得 JavaScript 在这方面与 WebAssembly 保持一致（Wasm 线性内存可以通过 [`WebAssembly.Memory.prototype.grow()`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory/grow) 调整大小）。
+
+出于安全原因，`SharedArrayBuffer` 无法缩小大小，只能增长。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:87]:
```suggestion
出于安全原因，`SharedArrayBuffer` 的大小无法缩小，只能增长。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 3]
```diff
@@ -1,98 +1,129 @@
...
+  const buffer = new ArrayBuffer(16);
+  myWorker.postMessage(buffer);
 }
 ```
 
-参阅 [Planned changes to shared memory](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer/Planned_changes)，以了解共享内存在浏览器中的变更计划（例如，Firefox 79）。
+在设置了这两个头后，`postMessage()` 不再为 `SharedArrayBuffer` 对象抛出错误，因此，跨线程共享内存现在可用。
 
-### 始终使用 new 运算符来构造 SharedArrayBuffer
+嵌套文档和专用 worker 线程也需要设置 [`Cross-Origin-Embedder-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) 头为同样的值。对于同源嵌套文档和子资源，不需要进行任何其他更改。同站（但跨域）嵌套文档和子资源需要设置 [`Cross-Origin-Resource-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) 头为 `same-site`。而它们的跨域（和跨站点）的对应部分也需要设置同样的头为 `cross-origin`。请注意，将 [`Cross-Origin-Resource-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) 头设置为除 `same-origin` 之外的任何值，都会使资源暴露于潜在的攻击中，比如[幽灵漏洞](https://zh.wikipedia.org/wiki/幽灵漏洞)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:63]:
```suggestion
嵌套文档和专用 worker 线程也需要将 [`Cross-Origin-Embedder-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) 标头设置为同样的值。对于同源嵌套文档和子资源，不需要进行任何其他更改。同站（但跨源）嵌套文档和子资源需要将 [`Cross-Origin-Resource-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) 标头设置为 `same-site`。而它们的跨源（和跨站点）的对应部分也需要将同样的标头设置为 `cross-origin`。请注意，将 [`Cross-Origin-Resource-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) 标头设置为除 `same-origin` 之外的任何值，都会使资源暴露于潜在的攻击中，比如[幽灵漏洞](https://zh.wikipedia.org/wiki/幽灵漏洞)。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 4]
```diff
@@ -1,98 +1,129 @@
...
 
-`SharedArrayBuffer` 必须使用 {{jsxref("Operators/new", "new")}} 运算符来构造。作为函数来调用 `SharedArrayBuffer()` 构造函数（如果不用 `new` 运算符），将会抛出 {{jsxref("TypeError")}} 异常。
+请注意，[`Cross-Origin-Opener-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy) 头部会限制你对弹出窗口引用的保留能力。两个顶级窗口上下文之间的直接访问基本上只在它们同源且携带相同的两个头及和对应值时才可行。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:65]:
```suggestion
请注意，[`Cross-Origin-Opener-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy) 标头会限制你对弹出窗口引用的保留能力。两个顶级窗口上下文之间的直接访问基本上只在它们同源且携带相同的两个标头（且具有相同的值）时才可行。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 5]
```diff
@@ -1,98 +1,129 @@
...
-```
+### API 可用性
 
-```js example-good
-var sab = new SharedArrayBuffer(1024);
-```
+根据是否采取了上述安全措施，各类内存共享 API 具有不同的可用性：
+
+- `Atomics` 对象总是可用的。
+- `SharedArrayBuffer` 对象在原则上始终可用，但遗憾的是，除非设置了前面提到的两个头，否则其在全局对象上的构造函数是隐藏的，这是为了兼容 web 内容。这个限制有望在未来被移除。尽管如此，仍然可以用 [`WebAssembly.Memory`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory) 来获取实例。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:72]:
```suggestion
- `SharedArrayBuffer` 对象在原则上始终可用，但遗憾的是，除非设置了前面提到的两个标头，否则其在全局对象上的构造函数是隐藏的，这是为了兼容 web 内容。这个限制有望在未来被移除。尽管如此，仍然可以用 [`WebAssembly.Memory`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory) 来获取实例。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 6]
```diff
@@ -1,98 +1,129 @@
...
+- 除非设置了上文提到的两个头，否则各种 `postMessage()` 的 API 在处理 `SharedArrayBuffer` 对象时会抛出异常。如果正确设置了这两个头，`Window` 对象和专用 worker 线程上的 `postMessage()` 都可以正常工作，并允许跨线程共享内存。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:73]:
```suggestion
- 除非设置了上文提到的两个标头，否则各种 `postMessage()` 的 API 在处理 `SharedArrayBuffer` 对象时会抛出异常。如果正确设置了这两个标头，`Window` 对象和专用 worker 线程上的 `postMessage()` 都可以正常工作，并允许跨线程共享内存。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 7]
```diff
@@ -1,98 +1,129 @@
...
+
+### WebAssembly 共享内存
+
+[`WebAssembly.Memory`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory) 对象可以通过设置 [`shared`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory/Memory#shared) 构造函数标志来创建。当这个标志设置为 `true` 时，构造出的 `Memory` 对象就像 `SharedArrayBuffer` 一样，可以通过 `postMessage()` 在 worker 线程之间共享，而且 `Memory` 对象的后备 [`buffer`](/zh-CN/docs/WebAssembly/JavaScript_interface/Memory/buffer) 是一个 `SharedArrayBuffer`。因此，上述关于在 worker 线程间共享 SharedArrayBuffer 的要求同样适用于共享 `WebAssembly.Memory`。
+
+WebAssembly Threads 提案还定义了一套新的[原子](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#atomic-memory-accesses)指令。就像 `SharedArrayBuffer` 及其方法始终可用（并且只有在设置了新头部的情况下，才允许线程间共享）一样，WebAssembly 原子指令也是始终可用的。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:79]:
```suggestion
WebAssembly Thread 提案还定义了一套新的[原子](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#atomic-memory-accesses)指令。就像 `SharedArrayBuffer` 及其方法始终可用（并且只有在设置了新标头的情况下，才允许线程间共享）一样，WebAssembly 原子指令也是始终可用的。
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 8]
```diff
@@ -1,98 +1,129 @@
...
+
+### 增长 SharedArrayBuffer
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:81]: 扩大/增大？

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 9]
```diff
@@ -126,8 +157,13 @@ gl.bufferData(gl.ARRAY_BUFFER, sab, gl.STATIC_DRAW);
 
 - {{jsxref("Atomics")}}
 - {{jsxref("ArrayBuffer")}}
-- [JavaScript typed arrays](/zh-CN/docs/Web/JavaScript/Typed_arrays)
+- [JavaScript 类型化数组](/zh-CN/docs/Web/JavaScript/Guide/Typed_arrays)指南
 - [Web Workers](/zh-CN/docs/Web/API/Web_Workers_API)
-- [parlib-simple](https://github.com/lars-t-hansen/parlib-simple)——一个提供同步和工作分配抽象的简单库。
-- [Shared Memory – a brief tutorial](https://github.com/tc39/ecmascript_sharedmem/blob/master/TUTORIAL.md)
-- [A Taste of JavaScript's New Parallel Primitives – Mozilla Hacks](https://hacks.mozilla.org/2016/05/a-taste-of-javascripts-new-parallel-primitives/)
+- [共享内存——简明教程](https://github.com/tc39/proposal-ecmascript-sharedmem/blob/main/TUTORIAL.md)在 TC39 ecmascript-sharedmem 提案中
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:162]:
```suggestion
- [共享内存——简明教程](https://github.com/tc39/proposal-ecmascript-sharedmem/blob/main/TUTORIAL.md)，TC39 ecmascript-sharedmem 提案
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 10]
```diff
@@ -126,8 +157,13 @@ gl.bufferData(gl.ARRAY_BUFFER, sab, gl.STATIC_DRAW);
...
+- [JavaScript's 新并发方法的初体验](https://hacks.mozilla.org/2016/05/a-taste-of-javascripts-new-parallel-primitives/) 在 hacks.mozilla.org（2016）
+- [COOP 和 COEP 的解释](https://docs.google.com/document/d/1zDlfvfTJ_9e8Jdc8ehuV4zMEu9ySMCiTGMS9y0GU92k/edit) 由 Chrome 团队撰写（2020）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:164]:
```suggestion
- [JavaScript's 新并发方法的初体验](https://hacks.mozilla.org/2016/05/a-taste-of-javascripts-new-parallel-primitives/)，hacks.mozilla.org（2016）
- [COOP 和 COEP 的解释](https://docs.google.com/document/d/1zDlfvfTJ_9e8Jdc8ehuV4zMEu9ySMCiTGMS9y0GU92k/edit)，由 Chrome 团队撰写（2020）
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 11]
```diff
@@ -126,8 +157,13 @@ gl.bufferData(gl.ARRAY_BUFFER, sab, gl.STATIC_DRAW);
...
+- {{HTTPHeader("Cross-Origin-Opener-Policy")}}
+- {{HTTPHeader("Cross-Origin-Embedder-Policy")}}
+- {{HTTPHeader("Cross-Origin-Resource-Policy")}}
+- [`crossOriginIsolated`](/zh-CN/docs/Web/API/crossOriginIsolated)
+- [Android Chrome 88 和桌面版 Chrome 92 中的 SharedArrayBuffer 更新](https://developer.chrome.com/blog/enabling-shared-array-buffer/) 在 developer.chrome.com（2021）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:169]:
```suggestion
- [Android Chrome 88 和桌面版 Chrome 92 中的 SharedArrayBuffer 更新](https://developer.chrome.com/blog/enabling-shared-array-buffer/)，developer.chrome.com（2021）
```

File: files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 12]
```diff
@@ -1,98 +1,129 @@
...
+- [`Cross-Origin-Embedder-Policy`](/zh-CN/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) 设置为 `require-corp` 或 `credentialless`（保护受害者免受你的源站点的影响）
 
-```plain
+```http
 Cross-Origin-Opener-Policy: same-origin
 Cross-Origin-Embedder-Policy: require-corp
 ```
 
-为检查跨域隔离是否生效，你可以检查 [`crossOriginIsolated`](/zh-CN/docs/Web/API/crossOriginIsolated) 属性在窗口和 `worker` 上下文中是否可用：
+为了验证跨域隔离是否生效，你可以测试 `window` 和 `worker` 上下文中是否存在 [`crossOriginIsolated`](/zh-CN/docs/Web/API/crossOriginIsolated) 属性：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/sharedarraybuffer/index.md:47]:
```suggestion
为了验证跨源隔离是否生效，你可以测试 window 和 worker 上下文中是否存在 [`crossOriginIsolated`](/zh-CN/docs/Web/API/crossOriginIsolated) 属性：
```

**Comment by JasonLamv-t** [comment]
> > Is it possible to consider translating `worker thread(s)` to `工作线程`？I noticed that some Chinese technical articles or blogs translated it as so, and the `System Worker Thread`, a similar concept which some different, in some [Microsoft documentation](https://learn.microsoft.com/zh-cn/windows-hardware/drivers/kernel/system-worker-threads).
> 
> I tend not to translate this word. As we don't have a generally recognized translation of "web worker".
> 
> If we use "工作线程", so how could we translate "web worker", should we use "web 工作者", is that mean a career that mainly involves in the web? It's a bit confusing.

Every translation should follow the origin and context, and I think sometimes context is more important than the content, only which can represent the writer's intention to the utmost extent. In my opinion, "web worker" should be translated to "Web 工作线程" if it means a concept in the "web worker API" or something related, "网络工作者" if it means a kind of career category. I don’t know if you’ve noticed, in some sentences of this document, "worker" should and was already translated to "worker 线程", which gives the article readability and reasonability. This kind of circumstance happens on other websites, e.g. [工作线程 on nodejs.cn](https://nodejs.cn/api/worker_threads.html). I am more curious about whether "工作线程" is accurate enough to adapt to various circumstances, considering the core concepts of "web worker" and "worker thread".

**Comment by yin1999** [comment]
Yes. We could tell that "a worker is a separate thread other than the main thread in web page". The emergence of workers is precisely to improve the performance of front-end web pages. However, the untranslated version - "Web Worker" - has been generally accepted by most of people. Whole W3C used to translate ["service workers" into "服务器线程"](https://www.w3.org/zh-hans/news/2015/service-workers-draft-published-3/), the translation of "Web Worker" is not given yet (even today).

There was also a discussion on its translation on telegram group:

image

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**